diff --git a/src/applications/people/storage/PhabricatorUserEmail.php b/src/applications/people/storage/PhabricatorUserEmail.php
index 4294601..fc12854 100644
--- a/src/applications/people/storage/PhabricatorUserEmail.php
+++ b/src/applications/people/storage/PhabricatorUserEmail.php
@@ -69,7 +69,7 @@ final class PhabricatorUserEmail extends PhabricatorUserDAO {
     // To this end, we're roughly verifying that there's some normal text, an
     // "@" symbol, and then some more normal text.
 
-    $email_regex = '(^[a-z0-9_+.!-]+@[a-z0-9_+:.-]+\z)i';
+    $email_regex = '(^[a-z0-9.!#$%&\'*+/=?\^_`{|}~\-]+@[a-z0-9_+:.\-]+$)i';
     if (!preg_match($email_regex, $address)) {
       return false;
     }
